app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+package com.linkedin.drelephant.exceptions;
+
+//Given a log string, analyzes and returns the relevant information!
+import org.apache.log4j.Logger;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AnalyzedLog {
+
+        private static final Logger logger = Logger.getLogger(AnalyzedLog.class);
+        private String state;
+        private ExceptionLoggingEvent exception;
+        private Set<String> list;
+        private String log;
+
+
+        public AnalyzedLog(String log){
+                Matcher m;
+                this.state = "";
+                this.exception = new ExceptionLoggingEvent();
+                this.list = new HashSet<String>();
+                Set<String> allMatches = new HashSet<String>();
+                this.log=log;
+
+                if (match("Flow \\'\\' is set to SUCCEEDED in [0-9]+ seconds").find()) {                                //Successful Azkaban flow log
+                        this.state = "SUCCEEDED";
+                }
+
+                else if (match("Setting flow \\'\\' status to FAILED in [0-9]+ seconds").find()) {                      //Failed Azkaban flow log
```

Comment:
Can't you get the succeeded/failed from the Azkaban rest api ? (Fetch a Flow Execution) You should use that instead of parsing the log. Reserve the parsing of logs only to get the exception.